test/src/vfs_test_helper.h

```diff
@@ -0,0 +1,443 @@
+   * @return Status OK if successful
+   */
+  virtual Status init(tiledb_ctx_t* ctx, tiledb_vfs_t* vfs);
+
+  /**
+   * Removes container if exists
+   *
+   * @param ctx The TileDB context.
+   * @param vfs The VFS object.
+   * @return Status OK if successful
+   */
+  virtual Status close(tiledb_ctx_t* ctx, tiledb_vfs_t* vfs);
+
+  /**
+   * Get the name of the filesystem's directory
+   *
+   * @return string directory name
+   */
+  virtual std::string temp_dir();
+
+ private:
+  /* ********************************* */
+  /*           ATTRIBUTES              */
+  /* ********************************* */
+
+  /** The directory prefix of the Azure filesystem. */
+  const std::string azure_prefix_;
+
+  /** The container name for the Azure filesystem. */
+  const std::string container;
+
+  /** The directory name of the Azure filesystem. */
+  const std::string azure_temp_dir_;
+
+  /** The directory name of the Azure filesystem. */
+  std::string temp_dir_;
+};
+
+/**
+ * This class provides support for
+ * the Windows or Posix (local) filesystem.
+ */
+class SupportedFsLocal : public SupportedFs {
+ public:
+#ifdef _WIN32
+  SupportedFsLocal()
+      : temp_dir_(tiledb::sm::Win::current_dir() + "\\tiledb_test\\")
+      , file_prefix_("") {
+  }
+#else
+  SupportedFsLocal()
+      : temp_dir_(tiledb::sm::Posix::current_dir() + "/tiledb_test/")
+      , file_prefix_("file://") {
+  }
+#endif
+
+  ~SupportedFsLocal() = default;
+  /* ********************************* */
+  /*               API                 */
+  /* ********************************* */
+
+  /**
+   * No-op
+   *
+   * @param config Configuration parameters
+   * @return Status OK if successful
+   */
+  virtual Status prepare_config(tiledb_config_t* config, tiledb_error_t* error);
+
+  /**
+   * No-op
+   *
+   * @param ctx The TileDB context.
+   * @param vfs The VFS object.
+   * @return Status OK if successful
+   */
+  virtual Status init(tiledb_ctx_t* ctx, tiledb_vfs_t* vfs);
+
+  /**
+   * No-op
+   *
+   * @param ctx The TileDB context.
+   * @param vfs The VFS object.
+   * @return Status OK if successful
+   */
+  virtual Status close(tiledb_ctx_t* ctx, tiledb_vfs_t* vfs);
+
+  /**
+   * Get the name of the filesystem's directory
+   *
+   * @return string directory name
+   */
+  virtual std::string temp_dir();
+
+  /**
+   * Get the name of the filesystem's file prefix
+   *
+   * @return string prefix name
+   */
+  std::string file_prefix();
+
+ private:
+  /* ********************************* */
+  /*           ATTRIBUTES              */
+  /* ********************************* */
+
+#ifdef _WIN32
+  /** The directory name of the Windows filesystem. */
+  std::string temp_dir_;
+
+  /** The file prefix name of the Windows filesystem. */
+  std::string file_prefix_;
+
+#else
+  /** The directory name of the Posix filesystem. */
+  std::string temp_dir_;
+
+  /** The file prefix name of the Posix filesystem. */
+  std::string file_prefix_;
+
+#endif
+};
+
+/**
```

Comment:
Similar to my comment in the .cc file, let's move these c-style functions to the top of this .h file.